auditbeat/module/auditd/audit_linux.go

```diff
@@ -154,21 +154,27 @@ func closeAuditClient(client *libaudit.AuditClient, log *logp.Logger) {
 	// Drain the netlink channel in parallel to Close() to prevent a deadlock.
 	// This goroutine will terminate once receive from netlink errors (EBADF,
 	// EBADFD, or any other error). This happens because the fd is closed.
-	go func() {
+	closeWaiter := &sync.WaitGroup{}
+	closeWaiter.Add(1)
+	go func(testClient *libaudit.AuditClient) {
 		for {
-			_, err := client.Netlink.Receive(true, discard)
+			_, err := testClient.Netlink.Receive(true, discard)
```

Comment:
Suggest
```
var wg sync.WaitGroup
wg.Add(1)
go func() {
    defer wg.Done()
    for {
        _, err := client.Netlink.Receive(true, discard)
```
with associated changes below.

rationale:
- hyperlocal labels don't need to be so heavy
- localise the wg
- the captured scope is just as safe as passing the parameter

Reply:
Yeah, agreed, that's a bit cleaner